store/postgres/src/writable.rs

```diff
@@ -716,6 +726,10 @@ impl Queue {
         }
     }
 
+    fn get_derived_ids(&self, key: &DerivedKey) -> Result<Option<DerivedEntityIds>, StoreError> {
+        self.store.get_dervied_ids(&key, BLOCK_NUMBER_MAX)
+    }
```

Comment:
The queue is a pipeline of entity operations going into the store. For example it may be that when processing block `N`, block `N - 1` has not yet been transacted into the store, but is queued. If you access `self.store`, the queue is being bypassed so writes in block `N - 1` would be ignored.

So this needs to take the modifications in the `queue` into account. It's a similar problem to the entity cache, it contains pending writes. This is tricky to handle but looking at the approach in `fn get_many` should be a good start.